disco/preprocess/select_timepoints2.py

```diff
@@ -5,154 +5,310 @@
 @author: ksedzro
 """
 
-# -*- coding: utf-8 -*-
-"""
-Created on Fri Nov 25 07:43:46 2022
-
-@author: ksedzro
-"""
+import filecmp
+import enum
+import logging
+import shutil
+import time
+from pathlib import Path
 
-import os
 import numpy as np
 import pandas as pd
-import time
 import opendssdirect as dss
 
-def load_feeder(path_to_master):
-    if os.path.exists(path_to_master):
-        dpath, fname = os.path.split(path_to_master)
-        newfname = fname.split('.')[0]
-        new_master = os.path.join(dpath, f"{newfname}_new.dss")
-        with open(path_to_master, 'r') as mr:
-            slines = mr.readlines()
-        lines = []
-        bk = False
-        for line in slines:
-            
-            if line.lower().startswith('solve'):
-                line = "Solve\n"
-                bk = True
-            lines.append(line)
-            if bk:
-                break
-        with open(new_master, 'w') as mw:
-            mw.writelines(lines)
-            
-    r = dss.run_command(f"redirect {new_master}")
-    if r != '':
-        print(f"PoenDSS error: {r}")
-        
-    
-def get_param_values(param_class, bus_data, category='demand'):
-    assert category in ['demand', 'generation']
-    
+
+logger = logging.getLogger(__name__)
+
+
+_DISALLOWED_OPEN_DSS_COMMANDS = ("export", "plot", "show")
+
+
+class CriticalCondition(enum.Enum):
+    """Possible critical conditions to use for time-point selection"""
+
+    MAX_DEMAND = "max_demand"
+    MIN_DEMAND = "min_demand"
+    MAX_GENERATION = "max_generation"
+    MAX_NET_GENERATION = "max_net_generation"
+
+
+class DemandCategory(enum.Enum):
+
+    LOAD = "load"
+
+
+class GenerationCategory(enum.Enum):
+
+    PV_SYSTEM = "pv_system"
+    STORAGE = "storage"
+
+
+class InvalidParameter(Exception):
+    """Raised when user input is invalid"""
+
+
+def load_feeder(path_to_master: Path, destination_dir: Path, fix_master_file: bool):
+    """Compile an OpenDSS circuit after first ensuring that time-series mode is disabled."""
+    if not path_to_master.exists():
+        raise FileNotFoundError(path_to_master)
+
+    if fix_master_file:
+        new_master = make_fixes_to_master(path_to_master)
+    else:
+        new_master = path_to_master
+    check_master_file(new_master)
+
+    try:
+        dss.Text.Command(f"redirect {new_master}")
+        shutil.copyfile(new_master, destination_dir / new_master.name)
+        logger.info("Redirected to %s", new_master)
+    finally:
+        if path_to_master != new_master:
+            new_master.unlink()
```

Comment:
Note that we delete any copies of the file we make in the user directory, but we do preserve a copy of what we used in the destination directory.

Reply:
Okay